### PR TITLE
Add a `period` parameter to the counter cache in remote write

### PR DESCRIPTION
**Pull request by gizas**
- Bug

## Proposed commit message

- WHAT: Added period parameter for counter cache in remote write dataset of prometheus
- WHY:  The rate calculation of counter metrics in Prometheus remote_write dataset was reporting zero values

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## How to test this PR locally

1. Checkout this branch
2. `cd packages/prometheus`
3. Run `elastic-package build` command
4. Initiate a new Elastic cluster `elastic-package stack up -d -v --version=8.14.0-SNAPSHOT` 
5. Login to Kibana and **install Prometheus Assets**
6. Install prometheus server with remote write in your kubernetes cluster
7. Configure elastic-agent
```yaml
inputs:
    - name: prometheus
      type: prometheus/metrics
      use_output: default
      meta:
        package:
          name: prometheus
          version: 1.15.1
      data_stream:
        namespace: default
      streams:
        - data_stream:
            dataset: prometheus.remote_write
            type: metrics
          metricsets:
            - remote_write
          host: '0.0.0.0'
          port: 9201
          rate_counters: true
          use_types: true
          period: 1m
```

## Related issues

- Relates #https://github.com/elastic/beats/issues/38458

## Screenshots

For my local kubernetes cluster (40m evidence for correct rate calculation with elastic-agent-standalone manifest)

Screenshot 2024-03-22 at 12 02 13 PM

### CODE REVIEW

File: packages/prometheus/changelog.yml [hunk 1]
```diff
@@ -1,4 +1,9 @@
 # newer versions go on top
+- version: "1.15.1"
+  changes:
+    - description: Add Period parameter for counter cache
+      type: bugfix
+      link: https://github.com/elastic/integrations/pull/9321
```
Comment (lalit-satapathy) [inline packages/prometheus/changelog.yml:6]: Nit: fix the PR number

**Review by lalit-satapathy — APPROVED** [review]
LGTM.

File: packages/prometheus/changelog.yml (hunk 1 above)
Comment (gizas) [reply packages/prometheus/changelog.yml]: Thank you!!!! Missed it, done

File: packages/prometheus/data_stream/remote_write/manifest.yml [hunk 2]
```diff
@@ -46,6 +46,14 @@ streams:
         multi: true
         required: false
         show_user: false
+      - name: period
+        type: text
+        title: Period
+        description: This is the timeout period of the counter cache used to calculate the counter rates
```
Comment (shmsr) [inline packages/prometheus/data_stream/remote_write/manifest.yml:52]:
```suggestion
        description: This is the timeout period for the counter cache, which is used to calculate counter rates.
```

File: packages/prometheus/changelog.yml [hunk 3]
```diff
@@ -1,4 +1,9 @@
 # newer versions go on top
+- version: "1.15.1"
+  changes:
+    - description: Add Period parameter for counter cache
```
Comment (shmsr) [inline packages/prometheus/changelog.yml:4]:
```suggestion
    - description: Add a "period" parameter to the counter cache.
```

**Review by shmsr — CHANGES_REQUESTED** [review]

**Review by ishleenk17 — APPROVED** [review]
Looks good!

**Review by shmsr — APPROVED** [review]

**Pull request merged**